client/deposits/list/test/index.js

```diff
@@ -122,4 +135,99 @@ describe( 'Deposits list', () => {
 
 		expect( tableSummary ).toHaveLength( 1 );
 	} );
+
+	describe( 'Download button', () => {
+		test( 'renders when there are one or more deposits', () => {
+			useDeposits.mockReturnValue( {
+				deposits: mockDeposits,
+				isLoading: false,
+			} );
+
+			const { queryByRole } = render( <DepositsList /> );
+			const button = queryByRole( 'button', { name: 'Download' } );
+
+			expect( button ).not.toBeNull();
+		} );
+
+		test( 'does not render when there are no deposits', () => {
+			useDeposits.mockReturnValue( {
+				deposits: [],
+				isLoading: false,
+			} );
+
+			const { queryByRole } = render( <DepositsList /> );
+			const button = queryByRole( 'button', { name: 'Download' } );
+
+			expect( button ).toBeNull();
+		} );
+	} );
+
+	describe( 'CSV download', () => {
+		beforeEach( () => {
+			useDeposits.mockReturnValue( {
+				deposits: mockDeposits,
+				depositsCount: 2,
+				isLoading: false,
+			} );
+			useDepositsSummary.mockReturnValue( {
+				depositsSummary: {
+					count: 2,
+					total: 5000,
+				},
+				isLoading: false,
+			} );
+		} );
+
+		afterEach( () => {
+			jest.resetAllMocks();
+		} );
+
+		afterAll( () => {
+			jest.restoreAllMocks();
+		} );
+
+		test( 'should render expected columns in CSV when the download button is clicked', () => {
+			const { getByRole } = render( <DepositsList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const expected = [
+				'',
+				'Date',
+				'Type',
+				'Amount',
+				'Status',
+				'"Bank account"',
+			];
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvHeaderRow = csvContent.split( os.EOL )[ 0 ].split( ',' );
+			expect( csvHeaderRow ).toEqual( expected );
+		} );
+
+		test( 'should match the visible rows', () => {
+			const { getByRole, getAllByRole } = render( <DepositsList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvLines = csvContent.split( os.EOL );
+			const cptRows = getAllByRole( 'row' );
+
+			expect( csvLines.length ).toEqual( cptRows.length );
+
+			const cptFirstDepositDate = cptRows[ 1 ].querySelector( 'td' )
+				.textContent;
+			const csvFirstDeposit = csvLines[ 1 ].split( ',' );
+			const csvFirstDepositDate = csvFirstDeposit[ 1 ];
+			const date = new Date( JSON.parse( csvFirstDepositDate ) );
+			const csvFirstDepositFormattedDate = dateI18n(
+				'M j, Y',
+				moment.utc( date.getTime() ).toISOString(),
+				true // TODO Change call to gmdateI18n and remove this deprecated param once WP 5.4 support ends.
+			);
+
+			expect( csvFirstDepositFormattedDate ).toEqual(
+				cptFirstDepositDate
+			);
```

Comment:
It sounds like this is just testing the date column for the first data row? 
IMHO, a more reasonable approach is to assert all fields of the first row between the current display and CSV file (which is similar to the way you assert `csvHeaderRow` in the previous test).

Also, instead of using JSON.parse, Date, dateI18n, etc, can we just simply compare a plain text such as `Jan 2, 2020`? 

Nitpick: `cpt` might be confusing with CPT = Custom Post Type in the WordPress context. If possible, I would suggest changing it to something else like `display`.

Reply:
> It sounds like this is just testing the date column for the first data row?
> IMHO, a more reasonable approach is to assert all fields of the first row between the current display and CSV file (which is similar to the way you assert `csvHeaderRow` in the previous test).

That indeed sounds like a more reasonable approach. I've accordingly refactored my test.

> Also, instead of using JSON.parse, Date, dateI18n, etc, can we just simply compare a plain text such as `Jan 2, 2020`?

This won't be possible as the exported date in CSV will need to be in a format that is more widely understood by both CSV editors and humans. Eg. 2021-08-30. Please see [this commit](https://github.com/Automattic/woocommerce-payments/pull/2778/commits/67b5e4fb6c7de3778dd482e606ed3174992c52e0) for related changes.

> Nitpick: `cpt` might be confusing with CPT = Custom Post Type in the WordPress context. If possible, I would suggest changing it to something else like `display`.

Right. @naman03malhotra found it confusing as well. Refactored now!